### DISCUSSION TITLE
How to make a window cover multiple screens in a multi-monitor situation

### DISCUSSION THREAD

**Discussion by lopo12123** [emilk/egui#3185] [Q&A] [answered]
I don't know if this is my requirement or a bug of eframe, I want my application window to cover all screens initially. On a single monitor it certainly works fine, but on multiple monitors it only seems to cover one screen at most, despite my large initial_window_size setting.

The code used for testing is available here: https://github.com/tiny-anchor/screen_capture
(In this program, I use another library to capture snapshot images of all screens, and I want to use eframe to create a window large enough to cover all monitors and render the images of each screen in the corresponding position. Hope someone can guide me to achieve this requirement)

**Comment by EvinceMoi** [comment, marked as answer]
I just ran into the exactly same issue as you did. With some digging inside the `eframe` crate ([window_builder](https://github.com/emilk/egui/blob/master/crates/eframe/src/native/epi_integration.rs#L162) and [largest_monitor_point_size](https://github.com/emilk/egui/blob/master/crates/eframe/src/native/epi_integration.rs#L203C14-L217)), I'd say that it was designed that way for some reason.

**Reply by lopo12123** [reply]
Thanks for your help let me know what the problem is

**Comment by LoganDark** [comment]
It could be that on Windows, it is seemingly impossible for a single window to be larger than a single monitor. You will probably have to use multiple windows. This is why multi-monitor mode is often a special case for games.

**Reply by lopo12123** [reply]
> it is seemingly impossible for a single window to be larger than a single monitor.

of course not! I can use either fltk or imgui to make a single window cover all the screens, but for some reason neither can achieve my personal needs (imgui does not have good support for compiling into node-api, and fltk seems to have some coordinate errors), so I came to egui to find a solution.

**Reply by lopo12123** [reply]
Or from the most intuitive example, you can use the mouse to drag the edge of a browser window to cover all your screens in a multi-screen situation. So Windows does not limit the window of a single application to be larger than the size of your screen.
